src/Misc/layoutroot/config.cmd

```diff
@@ -9,12 +9,29 @@ if defined VERBOSE_ARG (
 powershell.exe -NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command "$VerbosePreference = %VERBOSE_ARG% ; Get-ChildItem -LiteralPath '%~dp0' | ForEach-Object { Write-Verbose ('Unblock: {0}' -f $_.FullName) ; $_ } | Unblock-File | Out-Null ; Get-ChildItem -Recurse -LiteralPath '%~dp0bin', '%~dp0externals' | Where-Object { $_ -match '\.(ps1|psd1|psm1)$' } | ForEach-Object { Write-Verbose ('Unblock: {0}' -f $_.FullName) ; $_ } | Unblock-File | Out-Null"
 
 :run
-IF "%1"=="remove" (
-    "%~dp0bin\Agent.Listener.exe" unconfigure
-) ELSE (
-    "%~dp0bin\Agent.Listener.exe" configure %*
-    IF EXIST .Agent (attrib +h .Agent)
-    IF EXIST .Credentials (attrib +h .Credentials)
-    IF EXIST .Service (attrib +h .Service)
+
+IF "%~1" neq "remove" goto config
+shift
+set "args="
+:parse
+if "%~1" neq "" (
+    set args=%args% %1
+    shift
+    goto :parse
 )
+if defined args set args=%args:~1%
+"%~dp0bin\Agent.Listener.exe" unconfigure %args%
+goto end
+
+:config
+
+"%~dp0bin\Agent.Listener.exe" configure %*
+IF EXIST .Agent (attrib +h .Agent)
```

Comment:
It seems like setting the hidden attribute belongs in the Agent (when the files are written and where the names are well known). Otherwise if there is a need to do it here instead, then the paths should be resolved e.g. "%~dp0.Agent".